task-tracker-console-application/src/com/isa/user/Login.java

```diff
@@ -0,0 +1,134 @@
+package com.isa.user;
+
+import com.isa.MainMenu.MainMenu;
+
+import java.util.List;
+import java.util.Scanner;
+
+public class Login {
+    private static User loggedInUser;
+    private static final List<User> users = User.getTestUsers();
+    private static String inputLogin;
+
+    public static User getLoggedInUser() {
+        return loggedInUser;
+    }
+
+    public static void logOutUser() {
+        loggedInUser = null;
+    }
+
+    private static void displayLoginMessage() {
+        System.out.println("=====================================");
+        System.out.println("             ZALOGUJ SIĘ");
+        System.out.println("=====================================");
+        System.out.println("Podaj dane logowania lub wybierz \"0\", żeby powrócić do menu głównego.");
+    }
+
+    private static void logInUser(Scanner scanner) {
+        System.out.print("Podaj login: ");
+        validateLogin(scanner, 0);
+    }
+
+    private static void validateLogin(Scanner scanner, int incorrectLoginInputCount) {
+        inputLogin = scanner.nextLine();
+        if (inputLogin.equals("0")) {
+            goToMainMenu();
+        } else if (chosenUser() == null) {
+            incorrectLoginInputCount++;
+            System.out.println("Podany login nie istnieje.");
+            if (incorrectLoginInputCount > 1) {
+                System.out.println("Pamiętaj, że w każdej chwili możesz wrócić do menu głównego poprzez wybór \"0\".");
+            }
+
+            System.out.print("Podaj login ponownie: ");
+            validateLogin(scanner, incorrectLoginInputCount);
+        } else {
+            System.out.print("Podaj hasło: ");
+            validatePassword(scanner, 0);
+        }
+    }
+
+    private static User chosenUser() {
+        for (User u : users) {
+            if (u.getLogin().equals(inputLogin)) {
+                return u;
+            }
+        }
+        return null;
```

Comment:
chyba zamiast w takiego chamskiego nulla to poszedłym w stronę Optionala, żeby był zwracany z tej funkcji

Reply:
Dodałam Optionala, ale korzystam z .get(); nie wiem, jak to inaczej zrobić, i wolałabym nie korzystać za dużo z funkcjonalności, których jeszcze dobrze nie znam.

Reply:
dlatego na razie to zostawmy, a Optionale jeszcze sobie przegadamy 😃